### ISSUE TITLE
Rearranging output of `inspect_cat()` removes labels for `show_plot()`

### ISSUE THREAD

**Issue by RoelVerbelen** [alastairrushworth/inspectdf#27]
To improve spotting differences between datasets visually 
(especially when there are many columns) it would be helpful if one could sort the categorical columns by the Jensen–Shannon divergence.

The code below tries to do so but it seems to distort the labels on the y-axis. Also, in case the `jsd` column contains missing values, those variables are deleted from the graph.

``` r
library(inspectdf)
library(dplyr)
inspect_cat(starwars, starwars[1:20, ]) %>% 
  arrange(desc(jsd)) %>% 
  show_plot()
```

![](https://i.imgur.com/SkHFgNr.png)

<sup>Created on 2020-04-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by alastairrushworth** [comment]
Hi @RoelVerbelen.  I think the issue of using `arrange()` causing the downstream plot to have distorted label was fixed by a previous commit.  Please let me know if it hasn't.  

You also point out that if `jsd` is `NA` that the column comparison is dropped from the graphic.  I'm curious about the cases that might have caused this to happen - did you see a specific instance of this?  One case where this happens is where the column has different types in the two data frames, in which case I think dropping from the graphic could make sense.  But that may not be true in general...

Once again, thanks for the feedback and reports 😊

**Comment by alastairrushworth** [comment]
Reprex for the above....

``` r
library(inspectdf)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
inspect_cat(starwars, starwars[1:20, ]) %>% 
  arrange(desc(jsd)) %>% 
  show_plot()
```

![](https://i.imgur.com/FLuy6ES.png)

<sup>Created on 2020-06-25 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by RoelVerbelen** [comment]
Thanks again @alastairrushworth. In latest commit version, the labels are back but the arranging is completely ignored. Plotting is always done with the variable names sorted alphabetically. What would be the recommended approach to reorder the variables on the y-axis based on their corresponding `jsd` value?

I don't remember unfortunately how I ended up getting missing values for `jsd` before. I tried creating examples for the `starwars` data where one version of the data has 0 observations for some factor levels, thinking it would cause a division by 0 or log(0) in the `jsd` formula, but these scenarios seem to set it to 1. Also the trivial example where a factor is always missing resulted in a `jsd` of 1. I'll create a new issue if I run into missing values for `jsd` again.
